mfem build error with +rocm

```
/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/hip-5.5.1-36grozs3lkqmnph77fzw7tfbykoccwci/bin/hipcc -fPIC  -O3 -std=c++14 --offload-arch=gfx90a -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/hypre-2.\
30.0-sptx5zqbwmsaq7ga6kzptto6yshntpt3/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/openmpi-4.1.6-726wgocxnrdcpvhnyctkrwk5brwxpble/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/su\
perlu-dist-8.2.0-b7wlokhclgiyo5m3cs5bxjqbskmoyoyz/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/parmetis-4.0.3-357lo65nzpdjosuzjoqnpt2scrwu3jvf/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/\
gcc-11.4.0/metis-5.1.0-rv7yvjaonbvxlxr5pqvrgtqjlmrunnlr/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/sundials-6.6.1-r3237tcbjg3wnfg7q5qxzoje4ewgazbw/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04\
-zen4/gcc-11.4.0/petsc-3.20.0-rg55eg5xkqecpv2ed3l2tpdxxswlqu5o/include -I/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/hipsparse-5.5.1-hrdhigl6dssav4iifhwfzc42wfctyvjm/include -I/scratch/balay/spack/opt/spack/linux-ubu\
ntu22.04-zen4/gcc-11.4.0/zlib-ng-2.1.3-g3q24jtcwja4hbzlaqj5xpfppsd4kkgc/include  -c mesh/segment.cpp -o mesh/segment.o
In file included from linalg/handle.cpp:15:
In file included from linalg/petsc.hpp:45:
/scratch/balay/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.4.0/petsc-3.20.0-rg55eg5xkqecpv2ed3l2tpdxxswlqu5o/include/petscsystypes.h:531:16: fatal error: 'thrust/complex.h' file not found
      #include <thrust/complex.h>
               ^~~~~~~~~~~~~~~~~~
```
[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/12908610/spack-build-out.txt)

--- a/var/spack/repos/builtin/packages/mfem/package.py
+++ b/var/spack/repos/builtin/packages/mfem/package.py
@@ -923,6 +923,7 @@ class Mfem(Package, CudaPackage, ROCmPackage):
         if "+rocm" in spec:
             amdgpu_target = ",".join(spec.variants["amdgpu_target"].value)
             options += ["HIP_CXX=%s" % spec["hip"].hipcc, "HIP_ARCH=%s" % amdgpu_target]
+            hip_headers = HeaderList([])
             hip_libs = LibraryList([])
             # To use a C++ compiler that supports -xhip flag one can use
             # something like this:
@@ -933,7 +934,7 @@ class Mfem(Package, CudaPackage, ROCmPackage):
             #   hip_libs += find_libraries("libamdhip64", spec["hip"].prefix.lib)
             if "^hipsparse" in spec:  # hipsparse is needed @4.4.0:+rocm
                 hipsparse = spec["hipsparse"]
-                options += ["HIP_OPT=%s" % hipsparse.headers.cpp_flags]
+                hip_headers += hipsparse.headers
                 hip_libs += hipsparse.libs
                 # Note: MFEM's defaults.mk wants to find librocsparse.* in
                 # $(HIP_DIR)/lib, so we set HIP_DIR to be $ROCM_PATH when using
@@ -943,11 +944,16 @@ class Mfem(Package, CudaPackage, ROCmPackage):
                     options += ["HIP_DIR=%s" % env["ROCM_PATH"]]
                 else:
                     options += ["HIP_DIR=%s" % hipsparse["rocsparse"].prefix]
+            if "^rocthrust" in spec and not spec["hip"].external:
+                # petsc+rocm needs the rocthrust header path
+                hip_headers += spec["rocthrust"].headers
             if "%cce" in spec:
                 # We assume the proper Cray CCE module (cce) is loaded:
                 craylibs_path = env["CRAYLIBS_" + machine().upper()]
                 craylibs = ["libmodules", "libfi", "libcraymath", "libf", "libu", "libcsup"]
                 hip_libs += find_libraries(craylibs, craylibs_path)
+            if hip_headers:
+                options += ["HIP_OPT=%s" % hip_headers.cpp_flags]
             if hip_libs:
                 options += ["HIP_LIB=%s" % ld_flags_from_library_list(hip_libs)]
 

